src/OneExplorer/OneExplorer.ts

```diff
@@ -606,19 +606,39 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
    */
   delete(oneNode: OneNode): void {
     const isDirectory = (oneNode.node.type === NodeType.directory);
-    const title =
-        `Are you sure you want to delete '${path.parse(oneNode.node.path).base}'` + isDirectory ?
-        'and its contents?' :
-        '?';
-    const detail = `You can restore this file from the Trash.`;
-    const approval = 'Move to Trash';
-    const recursive = isDirectory ? true : false;
+
+    let recursive: boolean;
+    let title = `Are you sure you want to delete '${oneNode.node.name}'`;
+    if (isDirectory) {
+      title += ` and its contents?`;
+      recursive = true;
+    } else {
+      title += `?`;
+      recursive = false;
+    }
+
+    let detail: string|undefined;
+    let approval: string;
+    let useTrash: boolean;
+
+    if (vscode.env.remoteName === 'ssh-remote') {
```

Comment:
```suggestion
    if (vscode.env.remoteName) {
```

The `remoteName` is defined by extensions. There are many cases that can be included in this value. If it is remote, it can't use the trash, so it would be better to process it like this.

Reply:
I agree :-D
But I was not sure if `vscode.env.remoteName` would be undefined when it's run on local.

Reply:
Actually I cannot fully understand what the line means. Maybe I should find by `extensionKind`?

![image](https://user-images.githubusercontent.com/17171963/183877590-a3bca211-ef2c-4af2-9be6-a34d724c7128.png)

extensionKind in our package.json is 'workspace'.
![image](https://user-images.githubusercontent.com/17171963/183878543-3cf78836-e805-44d2-a881-8f08c54c608f.png)

Reply:
I also think it would be better to use `extensionKind` option. After I defined `extensionKind` as `ui` in our package.json file, I created a vsix package and installed it within the docker extension. Then, the `extensionKind` value was changed to `workspace`. But we need to investigate it more. :)

Reply:
Oh I just figured out what it means.

It concerns the case that our extension is installed in both local and remote. 
If so, the value will be always 'ssh-remote' even if we open the local workspace.
As our extension defines 'extensionKind' as 'workspace', it will run an extension only depending on where the workspace is (remote or local)

Reply:
It seems

(1) Case 1. Local host only
remoteName : undefined

(2) Case 2. Remote only
remoteName : ssh-remote or else, defined
extensionKind: workspace

(3) Case 3. Local extension and remote extension both installed
Because our exensionKind in 'workspace', it will run dependes on which directory we opens on vscode

When I opened the workspace in remote, extensionKind: workspace
When I opened the workspace in local window pc, extensionKind: ui

Reply:
What is the value in this case?

> (1) Case 1. Local host only
> remoteName : undefined

Reply:
@jyoungyun

Well I just uninstalled the remote installed one-vscode and tested in my window pc,
it's UI, as same as `Case 3-Local`.

I think that `~ the value is defined in all extension hosts (local and remote) in case a remote extension host exists.` is not about Case 3.

Reply:
------------
The conversation is moved to #1209

Reply:
Well now it looks like 'extensionKind: ui' means it's running on local PC, and 'extensionKind: workspace' means it's running on remote PC..